### PR TITLE
Refactor Eex Tokenizer

### PR DESCRIPTION
**Pull request by feliperenan**
This commit moves the line and column to the last argument to be handled as a metadata as you can see in the example below:

```elixir
    assert T.tokenize('foo', 1, 1, @opts) ==
             {:ok, [{:text, 'foo', %{line: 1, column: 1}}, {:eof, %{line: 1, column: 4}}]}
```

### CODE REVIEW

**Comment by josevalim** [comment]
Looks great to me, we just need to get CI green! :)

**Comment by josevalim** [comment]
Ah, don't forget to update the docs at the top of EEx.Tokenizer. :)

**Comment by feliperenan** [comment]
Yeah. I'm trying to look what I'm missing here to make it green.

Also, we are going to expose `token(contents, opts)`, right?

**Comment by josevalim** [comment]
We will expose EEx.tokenize(contents, opts), yes!

**Comment by feliperenan** [comment]
@josevalim I've updated the docs and exposed `Eex.tokenize/2`. I couldn't figure what is missing to get tests green though, I'm still investigating, but let me know if you have any clues - I guess it is something silly I haven't changed 🤔

File: lib/eex/lib/eex/compiler.ex [hunk 1]
```diff
@@ -195,16 +197,16 @@ defmodule EEx.Compiler do
...
     if only_spaces?(text) do
       look_ahead_middle(rest, start, contents ++ text)
     else
       nil
     end
   end
 
-  defp look_ahead_middle([{:middle_expr, line, _column, _, chars} | rest], _start, contents) do
-    {rest, line, contents ++ chars}
+  defp look_ahead_middle([{:middle_expr, chars, meta} | rest], _start, contents) do
```
Comment (josevalim) [inline lib/eex/lib/eex/compiler.ex:208]: Here is the bug:

```suggestion
  defp look_ahead_middle([{:middle_expr, _, chars, meta} | rest], _start, contents) do
```

:)

**Comment by josevalim** [comment]
@feliperenan can you also please change the tokenize errors to be of the shape `{:error, reason, meta}`?

File: lib/eex/lib/eex/compiler.ex (hunk 1 above)
Comment (feliperenan) [reply lib/eex/lib/eex/compiler.ex]: ohh man, I looked at it so many times but still couldn't find it 🙈 

Thanks ❤️

**Comment by feliperenan** [comment]
@josevalim Done!. Let me know if there is something else 🙌🏻

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**